modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/UndoRedo.js

```diff
@@ -48,53 +48,200 @@ export default function UndoRedo({
 		step: 0,
 	});
 
-	const handleUndoRedo = (newStep) => {
+	let descriptionInputComponent;
+	let friendlyURLInputComponent;
+	let titleInputComponent;
+
+	Liferay.componentReady(
+		`${portletNamespace}${META_FIELD_NAMES.description}`
+	).then((component) => {
+		descriptionInputComponent = component;
+	});
+
+	Liferay.componentReady(
+		`${portletNamespace}${META_FIELD_NAMES.friendlyURL}`
+	).then((component) => {
+		friendlyURLInputComponent = component;
+	});
+
+	Liferay.componentReady(`${portletNamespace}${META_FIELD_NAMES.title}`).then(
+		(component) => {
+			titleInputComponent = component;
+		}
+	);
+
+	const handleUndo = () => {
+		const newStep = step - 1;
 		const nextStep = history[newStep];
 
-		const titleInputComponent = Liferay.component(
-			`${portletNamespace}${META_FIELD_NAMES.title}`
+		if (nextStep.selectedLanguageId !== selectedLanguageId) {
+			const selectedLanguageIdInput = document.getElementById(
+				`${portletNamespace}languageId`
+			);
+
+			descriptionInputComponent
+				.get('translatedLanguages')
+				.values()
+				.map((lang) => {
+					if (
+						!nextStep.descriptionTranslatedLanguages.includes(lang)
+					) {
+						descriptionInputComponent
+							.get('translatedLanguages')
+							.remove(lang);
+						descriptionInputComponent.removeInputLanguage(lang);
+						descriptionInputComponent._updateTranslationStatus(
+							selectedLanguageId
+						);
+					}
+				});
+			descriptionInputComponent.selectFlag(nextStep.selectedLanguageId);
+			friendlyURLInputComponent
+				.get('translatedLanguages')
+				.values()
+				.map((lang) => {
+					if (
+						!nextStep.friendlyURLTranslatedLanguages.includes(lang)
+					) {
+						friendlyURLInputComponent
+							.get('translatedLanguages')
+							.remove(lang);
+						friendlyURLInputComponent.removeInputLanguage(lang);
+						friendlyURLInputComponent._updateTranslationStatus(
+							selectedLanguageId
+						);
+					}
+				});
+			friendlyURLInputComponent.selectFlag(nextStep.selectedLanguageId);
+			titleInputComponent
+				.get('translatedLanguages')
+				.values()
+				.map((lang) => {
+					if (!nextStep.titleTranslatedLanguages.includes(lang)) {
+						titleInputComponent
+							.get('translatedLanguages')
+							.remove(lang);
+						titleInputComponent.removeInputLanguage(lang);
+						titleInputComponent._updateTranslationStatus(
+							selectedLanguageId
+						);
+					}
+				});
+			titleInputComponent.selectFlag(nextStep.selectedLanguageId);
+
+			selectedLanguageIdInput.value = nextStep.selectedLanguageId;
+		}
+
+		descriptionInputComponent.updateInputLanguage(
```

Comment:
This last part is equal in both functions `handleUndo` and `handleRedo`. Could we extract it?